GraphConsoleAppV3/GraphRequests.cs

```diff
@@ -0,0 +1,1534 @@
+                    await user.UpdateAsync();
+                    Console.WriteLine("User {0} is successfully UNassigned an app (role).", user.DisplayName);
+
+                }
+            }
+
+            catch (Exception e)
+            {
+                Program.WriteError("\nError Assigning Direct Permission: {0} {1}", e.Message,
+                    e.InnerException != null ? e.InnerException.Message : "");
+            }
+
+        }
+
+        #endregion
+
+        private static async Task PrintDevices(IActiveDirectoryClient client)
+        {
+            #region Get Devices
+
+            //*********************************************************************************************
+            // Get a list of Mobile Devices from tenant
+            //*********************************************************************************************
+            Console.WriteLine("\nGetting Devices");
+            IPagedCollection<IDevice> devices = null;
+            try
+            {
+                devices = await client.Devices.ExecuteAsync();
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine("/nError getting devices {0} {1}", e.Message,
+                    e.InnerException != null ? e.InnerException.Message : "");
+            }
+
+            if (devices != null)
+            {
+                do
+                {
+                    List<IDevice> devicesList = devices.CurrentPage.ToList();
+                    foreach (IDevice device in devicesList)
+                    {
+                        if (device.ObjectId != null)
+                        {
+                            Console.WriteLine("Device ID: {0}, Type: {1}", device.DeviceId, device.DeviceOSType);
+                            IPagedCollection<IDirectoryObject> registeredOwners = device.RegisteredOwners;
+                            if (registeredOwners != null)
+                            {
+                                do
+                                {
+                                    List<IDirectoryObject> registeredOwnersList = registeredOwners.CurrentPage.ToList();
+                                    foreach (IDirectoryObject owner in registeredOwnersList)
+                                    {
+                                        Console.WriteLine("Device Owner ID: " + owner.ObjectId);
+                                    }
+                                    registeredOwners = await registeredOwners.GetNextPageAsync();
+                                } while (registeredOwners != null);
+                            }
+                        }
+                    }
+                    devices = await devices.GetNextPageAsync();
+                } while (devices != null);
+            }
+
+            #endregion
+        }
+
+        private static async Task CreateOAuth2Permission(IActiveDirectoryClient client, ServicePrincipal newServicePrincpal)
```

Comment:
Both have been extracted to their own method

---
In reply to: [80748731](https://github.com/Azure-Samples/active-directory-dotnet-graphapi-console/pull/37#discussion_r80748731) [](ancestors = 80748731)